server/server_test.go

```diff
@@ -949,6 +949,7 @@ func TestGrpcOverProxyProtocol(t *testing.T) {
 	var cfg Config
 	cfg.RegisterFlags(flag.NewFlagSet("", flag.ExitOnError))
 	cfg.ProxyProtocolEnabled = true
+	cfg.HTTPListenPort = 8080
```

Comment:
Github action runner complained about port permission in `make test` without this change

```
level=info msg="server listening on addresses" http=127.0.0.1:36861 grpc=127.0.0.1:42313
--- FAIL: TestGrpcOverProxyProtocol (0.00s)
    server_test.go:973: 
        	Error Trace:	/home/runner/work/dskit/dskit/server/server_test.go:9[73](https://github.com/grafana/dskit/actions/runs/9073285038/job/24930131832#step:6:74)
        	Error:      	Received unexpected error:
        	            	listen tcp :80: bind: permission denied
        	Test:       	TestGrpcOverProxyProtocol
```

Reply:
Could we just set this to 0 to have it choose a random port? Seems like the test doesn't care about the value of the HTTP port, and this will prevent conflicts for anyone with something else already listening on port 8080.